apis/python/src/tiledbsoma/_index_util.py

```diff
@@ -11,7 +11,7 @@
 from .options import SOMATileDBContext
```

Comment:
I realize this is not part of this PR, but I only just noticed this. We don't need to have a separate `_index_util.py` to solve this. `SOMATileDBContext` is only imported for typing, and the `typing` library has a special macro to deal with this circular import issue.

Here are some references for `TYPE_CHECKING`:
1. https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING
2. https://mypy.readthedocs.io/en/stable/runtime_troubles.html#typing-type-checking

Reply:
Lets add this change to this PR so the later reorganization is simple.  E.g.,

```
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .options import SOMATileDBContext
```